src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_zyx_fsv16_imad.cpp

```diff
@@ -453,7 +466,17 @@ ConvolutionKernelBase::DispatchData Convolution_kernel_b_fs_zyx_fsv16_imad::SetD
     DispatchData dispatchData;
     const auto& output = params.outputs[0];
     const auto& weights = params.weights;
-    auto block_params = GetBlockParams(params);
+
+    BlockParams block_params;
+
+    if (params.is_shape_agnostic) {
+        if (!shape_agnostic_block_params.is_initialized) {
+            const_cast<BlockParams&>(shape_agnostic_block_params) = GetBlockParams(params);
+        }
+        block_params = shape_agnostic_block_params;
+    } else {
+        block_params = GetBlockParams(params);
+    }
```

Comment:
Is it your intention to use **the same** block params for any shape agnostic kernel and node? Declaring `shape_agnostic_block_params` as a member of `kernel_b_fs_zyx_fsv16_imad` class would lead to reusing the same block params for all `kernel_b_fs_zyx_fsv16_imad` kernels after its initialization, despite differences in shape sizes, strides, filter sizes, etc

Also, this `const_cast<BlockParams&>` looks a little hacky. If the number of block configurations is predictable, probably we can have some kind of map for them? Or configure it during model loading and then pick the desired configuration at runtime?

BTW, `is_initialized` is never set to `true`

Reply:
Changed approach, please check